js/miso.js

```diff
@@ -27,7 +27,7 @@ function fetchJSON(url, method, body, headers, successful, errorful) {
     if (!response.ok) {
       throw new Error(response.statusText);
     }
-    return response.json();
+    return response.text();
```

Comment:
I think this means that we sometimes UTF-8 encode then decode unnecessarily. We should potentially use `.bytes()` instead.

There's no instance for `FromJSVal ByteString` unfortunately. There is `FromJSVal [Word8]` which is isomorphic, but obviously not the most performant data structure... Could do with some benchmarking. It'd be interesting to get some realistic benchmarks on strings anyway - I'm curious about how much `MisoString` helps when IME one is mostly interacting with libraries that use `Text`.

Reply:
We could try `JSON.parse(response.json())`, in the case of the `MIME` being `JSON`. Then we'd rely on the `FromJSVal Value` instance, since that seems to be faster than parsing. Haskell side it should be more performant.

For `OctetStream` we might want to consider trying `ArrayBuffer` (`.arrayBuffer()`, `.blob()` methods). There's also `.formData()` since I think `servant` supports that as well

Reply:
> Then we'd rely on the `FromJSVal Value` instance, since that seems to be faster than parsing, at least Haskell side it should be more performant.

I guess browsers probably have faster JSON parsers than Aeson, so yeah we could in effect do the `ByteString -> Aeson.Value` step there by using `.json()`. Whether it's worth the cost of doing FFI on a more complex object, I don't know. That part of JSaddle is opaque to me at this point. And anyway, it would have the downside mentioned in the previous comment, i.e. increased complexity.

Reply:
I'd say we find some way to operate on the raw bytes, and save special handling for JSON etc. as a potential future optimisation.